terracumber/git.py

```diff
@@ -62,6 +62,19 @@ def clone(self):
             # Maybe this is a tag
             self.repo.checkout('refs/tags/' + self.ref)
 
+    def get_last_commit(self):
+        """ Get last commit of the current branch """
+        try:
+            remote_branch = self.repo.lookup_branch("origin/" + self.ref, pygit2.GIT_BRANCH_REMOTE)
+            if remote_branch == None:
+                return None
+            branch_ref = self.repo.lookup_reference(remote_branch.name)
+            oid_commit = branch_ref.target
+            commit = self.repo[oid_commit]
+            return commit
+        except KeyError:
```

Comment:
maybe log it here and print:
```suggestion
        except (KeyError, pygit2.GitError) as e:
            logging.error(f"Error retrieving last commit: {e}")
            return None
```

Reply:
Yes, good idea

Reply:
Done